src/Compilers/VisualBasic/Portable/Binding/Binder_Expressions.vb

```diff
@@ -352,11 +352,12 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 boundArguments.Add(boundArgument)
 
                 Dim elementType = GetTupleFieldType(boundArgument, argumentSyntax, diagnostics, hasErrors)
-                elementTypes.Add(elementType)
-
                 If elementType Is Nothing Then
                     hasNaturalType = False
+                    elementType = ReclassifyExpression(boundArgument, diagnostics).Type
```

Comment:
Will try fetching inferred types from typeless expressions that are known to have inferred types. It is not that many. I just hoped to avoid that by reusing logic in ReclassifyExpression